### DISCUSSION TITLE
What does `eval-when-compile` block that requires dired, eshell, term, vterm do?

### DISCUSSION THREAD

**Discussion by daanturo** [magit/with-editor#112] [General]
```lisp
(eval-when-compile
  (progn (require 'dired nil t)
         (require 'eshell nil t)
         (require 'term nil t)
         (condition-case err
             (require 'vterm nil t)
           (error (message "Error(vterm): %S" err)))
         (require 'warnings nil t)))
```

At least the loading step, I don't think the above code block in `with-editor`'s source is necessary, we already have `declare-function` and `defvar` calls below to suppress compilation warning.

By getting rid of it, we can prevent problems like #85 from appearing from the first place.

Also under some circumstances when I have `global-git-commit-mode` hooked via `find-file-hook` (which pulls `with-editor`), opening a file with Emacs will immediately loads those 4 packages, which slows down the startup process significantly.

**Comment by tarsius** [comment]
Yeah, this seems strange. I've just removed it.
